### PR TITLE
Proxy 12  add validation

### PR DESCRIPTION
**Pull request by i-siz**
Validation added for existing queries.
Validation rules:
- asteroid request:
-- date - should be date in YYYY-MM-DD format less than now (including current date), not required
-- count_only - should be boolean value, not required
-- were_dangerous_meteors - should be boolean value, not required
- user request:
-- user_id - should be number, required
-- user_name - should be string, required
-- api_key - should be alphanumeric string with length 40 symbols, required

### CODE REVIEW

File: src/middlewares/validator-middleware.js [hunk 1]
```diff
@@ -0,0 +1,44 @@
+const validators = require('../validators')
+
+const supportedMethods = ["get", "post", "put", "patch", "delete"];
```
Comment (plutsenka) [inline src/middlewares/validator-middleware.js]: As you use `''` everywhere, it's better to use them here as well

File: src/validators/user-request-validator.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+const Joi = require('joi');
+
+const userRequestSchema = Joi.object({
+    user_id: Joi.number().required(),
+    user_name: Joi.string().alphanum().min(1).required(),
```
Comment (plutsenka) [inline src/validators/user-request-validator.js]: Probably just `.string()` is enough

File: src/middlewares/validator-middleware.js (hunk 1 above)
Comment (i-siz) [reply src/middlewares/validator-middleware.js]: Thanks. Updated.

File: src/validators/user-request-validator.js (hunk 2 above)
Comment (i-siz) [reply src/validators/user-request-validator.js]: I've done like this:
`user_name: Joi.string().required(),`
as I think that user name is still required.

**Review by plutsenka — APPROVED** [review]

**Pull request merged**